flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/sinks/StreamSelectTableSink.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sinks
+
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.core.execution.JobClient
+import org.apache.flink.streaming.api.datastream.{DataStream, DataStreamSink}
+import org.apache.flink.streaming.experimental.{CollectSink, SocketStreamIterator}
+import org.apache.flink.table.api.TableSchema
+import org.apache.flink.table.api.internal.SelectTableSink
+import org.apache.flink.table.types.DataType
+import org.apache.flink.types.Row
+
+import java.net.InetAddress
+import java.util
+
+/**
+  * A [[SelectTableSink]] for streaming select job.
+  *
+  * <p><strong>NOTES:</strong> This is a temporary solution,
+  * once FLINK-14807 is finished, the implementation should be changed.
```

Comment:
We'll leave the select in old planner untouched. FLINK-14807 only optimize select in blink planner.